content/altinn-api/clients/_index.md

```diff
@@ -0,0 +1,201 @@
+
+XML file iteration and instanceOwnerLookup creation:
+
+```c#
+foreach (string xmlFilePath in xmlFilePaths)
+            {
+                xmlFileName = xmlFilePath.Split("\\").Last();
+
+                // The person number is the XML filename
+                personNumber = xmlFileName.Split(".").First();
+
+                Instance instanceTemplate = new Instance()
+                {
+                    InstanceOwnerLookup = new InstanceOwnerLookup()
+                    {
+                        PersonNumber = personNumber,
+                    }
+                };
+```
+
+Multipart content creation:
+
+```c#
+MultipartFormDataContent content = new MultipartContentBuilder(instanceTemplate)
+                .AddDataElement("default", new FileStream(xmlFilePath, FileMode.Open), "application/xml")
+                .Build();
+```
+
+*We have made a helper class, `MultipartContentBuilder`, to easily add new data elements to the multipart in case several attachements are to be used for prefill.*
+
+In our example we have used an element type "`default`" and content type "`application/xml`". These two must be declared in the application metadata in order to store the files. The same goes for "`instance`" and "`application/json`" to create the instance, which is added within the constructor of `MultipartContentBuilder`:
+
+```c#
+public class MultipartContentBuilder
+    {
+        private MultipartFormDataContent builder;
+
+        public MultipartContentBuilder(Instance instanceTemplate)
+        {
+            builder = new MultipartFormDataContent();
+            if (instanceTemplate != null)
+            {
+                StringContent instanceContent = new StringContent(JsonConvert.SerializeObject(instanceTemplate), Encoding.UTF8, "application/json");
+
+                builder.Add(instanceContent, "instance");
+            }
+        }
+```
+
+The C# code then generates the following multipart information to be sent with the request to the Altinn API:
+
+```http
+Content-Type: multipart/form-data; boundary="abcdefg"
+
+--abcdefg
+Content-Type: application/json; charset=utf-8
+Content-Disposition: form-data; name="instance"
+
+{
+  "instanceOwnerLookup":  { "personNumber": "01036800298" },
+  "createdDateTime": "0001-01-01T00:00:00"
+}
+--abcdefg
+Content-Type: application/xml
+Content-Disposition: form-data; name="default"
+
+<?xml version="1.0" encoding="UTF-8" ?>
+<testcase>
+    <testperson>
+        Test person 1
+    </testperson>
+</testcase>
+--abcdefg--
+```
+
+Here we see the C# codeline to post the multipart content to the the Altinn API, and to retrieve the response:
+
+```c#
+HttpResponseMessage response = client.PostAsync(requestUri, content).Result;
+```
+
+The Altinn API processes this multipart by iterating over each content part separated by the boundary. This way we can do **one** post request to the server to achieve multiple upload of data files in one application instance.
```

Comment:
Replace this paragraph with deserialization (newtonsoft)

Reply:
Fixed.